app/models/concerns/sequenced.rb

```diff
@@ -12,6 +12,7 @@ module Sequenced
 
     def ensure_sequential_id
       return if sequential_id.present?
+      return if instance_of?(::Invoice) && !status_changed_to_finalized?
```

Comment:
Not sure that coupling this concern with the Invoice model is a good idea.

What about something like:
```ruby
...
  return unless should_assign_sequential_id?
...

def should_assign_sequential_id?
  true
end
```

And in the Invoice model:

```ruby
def should_assign_sequential_id?
  status_changed_to_finalized?
end
```

Reply:
Nice, Thanks for suggestion! Changed!